### PR TITLE
server/balancer: add scheduler and simplify schedule configs

### PR DESCRIPTION
**Pull request by huachaohuang**
The original balance configs are complicated and some of them are very
confusing, now we simplify them to make it more easy to understand.

We replace the concept of "balance" with "schedule". Balance means to
keep the resource balance, but we will introduce different kinds of
strategies in the future which may not balance, and balance is just one
kind of strategies to schedule resources between stores

### CODE REVIEW

File: server/balancer.go [hunk 1]
```diff
@@ -14,241 +14,208 @@
...
 
-// Balance tries to select a store region to do balance.
-// The balance type is follower balance.
-func (cb *capacityBalancer) Balance(cluster *clusterInfo) (float64, *balanceOperator, error) {
-	region, source, target := scheduleStorage(cluster, cb.selector)
-	if region == nil {
-		return 0, nil, nil
-	}
+func (l *leaderBalancer) GetResourceKind() ResourceKind {
+	return l.kind
```
Comment (siddontang) [inline server/balancer.go]: seem that the kind is fixed for leader and storage balancer, why not returning the const var directly?

**Comment by siddontang** [comment]
Cool, LGTM

**Comment by huachaohuang** [comment]
PTAL @overvenus

File: server/filter.go [hunk 2]
```diff
@@ -62,77 +62,78 @@ func (f *excludedFilter) FilterTarget(store *storeInfo) bool {
...
-func newCapacityFilter(cfg *BalanceConfig) *capacityFilter {
-	return &capacityFilter{cfg: cfg}
+func (f *regionCountFilter) FilterSource(store *storeInfo) bool {
+	return uint64(store.stats.RegionCount) < f.opt.GetMinRegionCount()
 }
 
-func (cf *capacityFilter) FilterSource(store *storeInfo) bool {
-	return store.storageRatio() <= cf.cfg.MinCapacityUsedRatio
+func (f *regionCountFilter) FilterTarget(store *storeInfo) bool {
+	return false
```
Comment (huachaohuang) [inline server/filter.go]: This filter is used to filter source stores with regions fewer than `MinRegionCount`, it doesn't filter target stores, so it just returns false.

**Pull request merged**